core/js/src/files/FileSystemJs.kt

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.io.files
+
+import kotlinx.io.IOException
+
+internal val fs: dynamic
+    get(): dynamic {
+        return try {
+            js("require('fs')")
+        } catch (t: Throwable) {
+            null
+        }
+    }
+
+internal val os: dynamic
+    get(): dynamic {
+        return try {
+            js("require('os')")
+        } catch (t: Throwable) {
+            null
+        }
+    }
+
+private class JsFileSystem : SystemFileSystemImpl() {
+    companion object {
+        val Instance = JsFileSystem()
```

Comment:
why can't we just make `JsFileSystem` an object?

Reply:
It can for sure, thanks!